src/Stratis.Bitcoin/P2P/ISelfEndpointTracker.cs

```diff
@@ -7,6 +11,18 @@ namespace Stratis.Bitcoin.P2P
     /// </summary>
     public interface ISelfEndpointTracker
     {
+        /// <summary>Update external IP address and peer score of the node.</summary>
+        /// <param name="ipEndPoint">The endpoint to add.</param>
+        /// <param name="suppliedEndpointIsFinal">Whether the ipEndPoint supplied should be marked final on the endpoint tracker.</param>
+        /// <param name="ipEndPointPeerScore">Peer score of the ipEndPoint supplied.  Default value of 1.</param>
+        void UpdateAndAssignMyExternalAddress(IPEndPoint ipEndPoint, bool suppliedEndpointIsFinal, int ipEndPointPeerScore = 0);
+
+        /// <summary>External IP address of the node.</summary>
+        IPEndPoint MyExternalAddress { get; }
+
+        /// <summary>Peer score of external IP address of the node.</summary>
```

Comment:
Now we can remove this one from the interface completely and it will become just a private property of the self endpoint tracker implementation